### PR TITLE
Dynamic imports of downloaders and builders

### PR DESCRIPTION
**Pull request by khardix**
Judging from the `# TODO: Use reflection.` comments, the current mechanism for loading downloaders and builders was a temporary solution.

I went ahead and tweaked the importing functions in `.utils`, so they now properly provide the requested class, and switched the `Base{Builder,Downloader}.get_instance` methods to utilize them.

As a bonus, I wrote [Napoleon](http://www.sphinx-doc.org/en/1.5.5/ext/napoleon.html)-style docstrings for all affected functions/methods.

### CODE REVIEW

File: rpmlb/builder/base.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 import retrying
 
+from .. import utils
+
 # from rpmlb import utils
```
Comment (hroncok) [inline rpmlb/builder/base.py]: Maybe also remove that in  the process?

File: rpmlb/builder/base.py [hunk 2]
```diff
@@ -16,29 +18,23 @@ class BaseBuilder:
...
-            from rpmlb.builder.dummy import DummyBuilder
-            instance = DummyBuilder()
-        else:
-            raise ValueError('name is invalid.')
+    @staticmethod
+    def get_instance(name):
+        """Instantiate named builder.
+
+        Keyword arguments:
+            name (str): The name of the requested builder.
```
Comment (hroncok) [inline rpmlb/builder/base.py]: If you already know this takes str, maybe use type annotation in function definition instead?

File: rpmlb/builder/base.py (hunk 1 above)
Comment (khardix) [inline rpmlb/builder/base.py]: Ah, missed that comment :) I will remove it.

File: rpmlb/builder/base.py (hunk 2 above)
Comment (khardix) [inline rpmlb/builder/base.py]: Good point, I forgot we are not on Py2-compat mode any more 🍾 .

**Pull request merged**